spec/lib/zendesk_apps_tools/deploy_spec.rb

```diff
@@ -44,23 +37,48 @@ def get_value_from_stdin(text)
     end
 
     context 'user inputs an app name that is in api response' do
-      define_method(:mock_methods_and_api) do |subject|
+      it 'returns app id 22 for notification_app' do
+        subject = mocked_instance_methods_and_api('notification_app')
         allow(subject).to receive_message_chain(:cache, :save)
-        allow(subject).to receive(:say_status)
-        allow(subject).to receive_message_chain(:connection, :get, :body) { api_response }
-      end
 
-      it 'returns app id 22 for notification_app' do
-        subject = subject_class.new('notification_app')
-        mock_methods_and_api(subject)
         expect(subject.find_app_id).to eq(22)
       end
 
       it 'returns app id 99 for notification_app' do
-        subject = subject_class.new('time_tracking_app')
-        mock_methods_and_api(subject)
+        subject = mocked_instance_methods_and_api('time_tracking_app')
+        allow(subject).to receive_message_chain(:cache, :save)
+
         expect(subject.find_app_id).to eq(99)
       end
     end
   end
+
+  describe '#check_job' do
+    let(:failed_api_response_body) { { 'status' => 'failed', 'message' => 'You failed!' }.to_json }
+    let(:completed_api_response_body){ { 'status' => 'completed', 'app_id' => 2}.to_json }
```

Comment:
Is `app_id` required here? Also perhaps make these two responses look the same except for the status code?

Reply:
App id is not required. Good catch.

About making the two responses look the same,
the failed message from response.body is captured and logged on the CLI, where as the one for completed isn't used anywhere in the code.

I'll have to go with this
```
let(:failed_api_response_body) { { 'status' => 'failed', 'message' => 'You failed!' }.to_json }

let(:completed_api_response_body){ { 'status' => 'completed'}.to_json } }
```